runtime/src/test/java/com/flipkart/flux/dao/StateMachinesDAOTest.java

```diff
@@ -13,56 +13,84 @@
+import com.flipkart.flux.runner.GuiceJunit4Runner;
 import junit.framework.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
 
-import javax.transaction.Transactional;
+import javax.inject.Inject;
+import java.util.HashSet;
+import java.util.Set;
 
 /**
  * <code>StateMachinesDAOTest</code> class tests the functionality of {@link StateMachinesDAO} using JUnit tests.
  * @author shyam.akirala
  * @author kartik.bommepally
  */
+@RunWith(GuiceJunit4Runner.class)
 public class StateMachinesDAOTest {
 
-    private Injector injector;
+    @Inject
+    @Rule
+    public DbClearRule dbClearRule;
+
+    @Inject
+    StateMachinesDAO stateMachinesDAO;
+
+    @Inject
+    StatesDAO statesDAO;
+
+    @Inject
+    StateMachinePersistenceService<DummyEventData> stateMachinePersistenceService;
 
     @Before
-    public void setup() {
-        injector = Guice.createInjector(new ConfigModule(), new HibernateModule());
-    }
+    public void setup() {}
 
     @Test
-    @Transactional
     public void createSMTest() {
-        StateMachinesDAO stateMachinesDAO = injector.getInstance(StateMachinesDAO.class);
-        StatesDAO statesDAO = injector.getInstance(StatesDAO.class);
-        DummyTask task = new DummyTask();
-        DummyOnEntryHook onEntryHook = new DummyOnEntryHook();
-        DummyOnExitHook onExitHook = new DummyOnExitHook();
-        State<DummyEventData> state1 = statesDAO.create(new State<DummyEventData>(2L, "state1", "desc1", onEntryHook, task, onExitHook, 3L, 60L));
-        State<DummyEventData> state2 = statesDAO.create(new State<DummyEventData>(2L, "state2", "desc2", null, null, null, 2L, 50L));
-        Set<State<DummyEventData>> states = new HashSet<State<DummyEventData>>();
+        DummyTask<DummyEventData> task = new DummyTask<DummyEventData>();
+        DummyOnEntryHook<DummyEventData> onEntryHook = new DummyOnEntryHook<DummyEventData>();
+        DummyOnExitHook<DummyEventData> onExitHook = new DummyOnExitHook<DummyEventData>();
+        State<DummyEventData> state1 = new State<DummyEventData>(2L, "state1__", "desc1", onEntryHook, task, onExitHook, 3L, 60L);
+        State<DummyEventData> state2 = new State<DummyEventData>(2L, "state2__", "desc2", null, task, null, 2L, 50L);
+        Set<State<DummyEventData>> states = new HashSet<>();
         states.add(state1);
         states.add(state2);
-        StateMachine<DummyEventData> stateMachine = new StateMachine<DummyEventData>(2L, "test_name", "test_desc", states);
+        StateMachine<DummyEventData> stateMachine = new StateMachine<>(2L, "SM_name_", "SM_desc_", states);
         String savedSMId = stateMachinesDAO.create(stateMachine).getId();
 
         StateMachine stateMachine1 = stateMachinesDAO.findById(savedSMId);
-        Assert.assertNotNull(stateMachine1);
+        Assert.assertEquals(stateMachine, stateMachine1);
+    }
+
+    /** Test creation of State machine from state machine definition */
```

Comment:
This test belongs to StateMachinePersistenceService. 
In fact, I would suggest we make one Resource level test for this.
Once done, we can also remove the createSMTest above